### DISCUSSION TITLE
Creating a Navigation Control with Animations

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#9066] [Q&A]
For a single-view app, let's say I want to create navigation.

I can create an empty UserControl that gets set as the main view, and then set the Content property to navigate. I guess. Need to test.

Now... what if I want some animation during page navigation, how would I do that?

I'd have a library that ends up with a NavigationRoot usercontrol and a NavigationManager control that keeps the navigation history.

**Comment by Mrxx99** [comment]
Take a look at https://docs.avaloniaui.net/docs/controls/transitioningcontentcontrol
